py_gtfs_rt_ingestion/ingest.py

```diff
@@ -106,56 +105,49 @@ def main(files: list[str]) -> None:
     except KeyError as e:
         raise ArgumentException("Missing S3 Bucket environment variable") from e
 
-    logging.info(
-        "Creating pool with %d threads, %d cores available",
-        POOL_SIZE,
-        os.cpu_count(),
-    )
-
-    # list of files to move to error bucket to be inspected and processed later
-    failed_ingestion = []
+    archive_files = []
+    error_files = []
 
     try:
-        config = Configuration(filename=files[0])
-
-        pa_table = pa.table(config.empty_table(), schema=config.export_schema)
-
-        with ThreadPoolExecutor(max_workers=POOL_SIZE) as executor:
-            for result in executor.map(
-                lambda x: gz_to_pyarrow(*x), [(f, config) for f in files]
-            ):
-                if isinstance(result, pa.Table):
-                    pa_table = pa.concat_tables([pa_table, result])
-                else:
-                    failed_ingestion.append(result)
-
-        logging.info(
-            "Completed converting %d files with config %s",
-            len(files),
-            config.config_type,
-        )
-
-        logging.info("Writing Table to %s", export_bucket)
+        config_type = ConfigType.from_filename(files[0])
+        converter: Converter
+        if config_type.is_gtfs():
+            converter = GTFSConverter(config_type)
+        elif config_type.is_gtfs_rt():
+            converter = GTFSRTConverter(config_type)
+        else:
+            raise NoImplException(f"No Converter for Config Type {config_type}")
+
+        # filesystem to use when writing parquet files
         s3_filesystem = fs.S3FileSystem()
-        pq.write_to_dataset(
-            table=pa_table,
-            root_path=os.path.join(export_bucket, str(config.config_type)),
-            filesystem=s3_filesystem,
-            partition_cols=["year", "month", "day", "hour"],
-        )
 
-    except (ConfigTypeFromFilenameException, NoImplException) as e:
+        for s3_prefix, table in converter.convert(files):
+            s3_path = os.path.join(export_bucket, s3_prefix)
+            logging.info("Writing Table to %s", s3_path)
+            pq.write_to_dataset(
+                table=table,
+                root_path=s3_path,
+                filesystem=s3_filesystem,
+                partition_cols=converter.partition_cols,
+            )
+
+        archive_files = converter.archive_files
+        error_files = converter.error_files
```

Comment:
convert files to `(s3_prefix, table)` tuples and write them out. then grab error and archive filenames.

NOTE: this is a for loop because `GTFSConverter` is generating a table for each csv in the gtfs zip file.